### PR TITLE
contracts: Improve documentation of lazy deletion config

### PR DESCRIPTION
**Pull request by athei**
Just added some better documentation for two items after receiving some questions about them.

### CODE REVIEW

File: frame/contracts/src/lib.rs [hunk 1]
```diff
@@ -255,10 +255,29 @@ pub mod pallet {
 		type CallStack: smallvec::Array<Item = Frame<Self>>;
 
 		/// The maximum number of tries that can be queued for deletion.
+		///
+		/// When a contract is deleted by calling `seal_terminate` it becomes inaccessible
+		/// immediatly but the deletion of the storage item it has accumulated is performaned
+		/// later. The contract address is put into the deletion queue. This defines how many
```
Comment (sacha-l) [inline frame/contracts/src/lib.rs:261]:
```suggestion
		/// immediately but the deletion of the storage item it has accumulated is performed
		/// later. The contract address is put into the deletion queue. This defines how many
```

File: frame/contracts/src/lib.rs [hunk 2]
```diff
@@ -255,10 +255,29 @@ pub mod pallet {
...
+		/// contracts can be queued up at the same time. If that limit is reached `seal_terminate`
+		/// will fail. The action then must be retried in a later block.
+		///
+		/// The reason for limiting the queue depth are:
```
Comment (sacha-l) [inline frame/contracts/src/lib.rs:265]:
```suggestion
		/// The reasons for limiting the queue depth are:
```

File: frame/contracts/src/lib.rs [hunk 3]
```diff
@@ -255,10 +255,29 @@ pub mod pallet {
...
+		/// 2. The queue is stored in a vector and needs to be decoded as a whole when reading
+		///		it at the end of each block. Longer queues take more weight to decode and hence
+		///		limit the amount of items that can be deleted per block.
 		#[pallet::constant]
 		type DeletionQueueDepth: Get<u32>;
 
 		/// The maximum amount of weight that can be consumed per block for lazy trie removal.
+		///
+		/// The amount of weight that is dedicated per block to work on the deletion queue. Larger
+		/// values allow more trie keys to be deleted in each each block but reduce the amount of
```
Comment (sacha-l) [inline frame/contracts/src/lib.rs:278]:
```suggestion
		/// values allow more trie keys to be deleted in each block but reduce the amount of
```

**Review by sacha-l — COMMENTED** [review]
A couple easy edits caught my eye when I saw this. 😃

File: frame/contracts/src/lib.rs [hunk 4]
```diff
@@ -255,10 +255,29 @@ pub mod pallet {
 		type CallStack: smallvec::Array<Item = Frame<Self>>;
 
 		/// The maximum number of tries that can be queued for deletion.
+		///
+		/// When a contract is deleted by calling `seal_terminate` it becomes inaccessible
+		/// immediately but the deletion of the storage item it has accumulated is performed
```
Comment (ascjones) [inline frame/contracts/src/lib.rs:260]:
```suggestion
		/// immediately, but the deletion of the storage item it has accumulated is performed
```

File: frame/contracts/src/lib.rs [hunk 5]
```diff
@@ -255,10 +255,29 @@ pub mod pallet {
 		type CallStack: smallvec::Array<Item = Frame<Self>>;
 
 		/// The maximum number of tries that can be queued for deletion.
```
Comment (ascjones) [inline frame/contracts/src/lib.rs:257]: Slightly ambiguous as "tries" could possibly be interpreted as "attempts" on first reading. Maybe fully qualify e.g. "contract storage tries"

**Review by ascjones — APPROVED** [review]
LGTM

File: frame/contracts/src/lib.rs (hunk 5 above)
Comment (athei) [reply frame/contracts/src/lib.rs]: Good point. Changed the wording.

**Review by HCastano — APPROVED** [review]

**Comment by athei** [comment]
bot merge

**Pull request merged**